WreckEngine/Source.cpp

```diff
@@ -12,166 +12,174 @@
-	//double spf = 1.0 / FPS;
-	//while (dt < spf) {
-	//	currFrame = glfwGetTime();
-	//	dt += currFrame - prevFrame;
-	//	prevFrame = currFrame;
-	//}
-
-	runningAvgDelta -= runningAvgDelta / samples;
-	runningAvgDelta += dt / samples;
-	auto title = std::to_string(fpsMode ? 1.0 / runningAvgDelta : runningAvgDelta * 1000.0);
-	auto decimal = title.find('.');
-	if (title.length() - decimal > 3) title = title.erase(decimal + 3);
-	title += fpsMode ? " FpS" : " MSpF";
-	glfwSetWindowTitle(Window::window, title.c_str());
-
-	Mouse::update();
-
-	bool alt = Window::getKey(GLFW_KEY_RIGHT_ALT) == GLFW_PRESS || Window::getKey(GLFW_KEY_LEFT_ALT) == GLFW_PRESS;
-	if (alt) {
-		if (Window::getKey(GLFW_KEY_0) == GLFW_PRESS)
-			fpsMode = !fpsMode;
-		if (Window::getKey(GLFW_KEY_EQUAL) == GLFW_PRESS)
-			FPS += (FPS < 5) ? 1 : ((FPS < 20) ? 5 : ((FPS < 60) ? 10 : 0));
-		if (Window::getKey(GLFW_KEY_MINUS) == GLFW_PRESS)
-			FPS -= (FPS > 20) ? 10 : ((FPS > 5) ? 5 : ((FPS > 1) ? 1 : 0));
-	}
-
-	game->update(dt);
+    auto currFrame = glfwGetTime();
+    auto dt = currFrame - prevFrame;
+    prevFrame = currFrame;
+    //need to separate drawing and update pipelines
+    //double spf = 1.0 / FPS;
+    //while (dt < spf) {
+    //  currFrame = glfwGetTime();
+    //  dt += currFrame - prevFrame;
+    //  prevFrame = currFrame;
+    //}
+
+    runningAvgDelta -= runningAvgDelta / samples;
+    runningAvgDelta += dt / samples;
+    auto title = std::to_string(fpsMode ? 1.0 / runningAvgDelta : runningAvgDelta * 1000.0);
+    auto decimal = title.find('.');
+    if (title.length() - decimal > 3) title = title.erase(decimal + 3);
+    title += fpsMode ? " FpS" : " MSpF";
+    glfwSetWindowTitle(Window::window, title.c_str());
+
+    Mouse::update();
+
+    bool alt = Window::getKey(GLFW_KEY_RIGHT_ALT) == GLFW_PRESS || Window::getKey(GLFW_KEY_LEFT_ALT) == GLFW_PRESS;
+    if (alt) {
+        if (Window::getKey(GLFW_KEY_0) == GLFW_PRESS)
+            fpsMode = !fpsMode;
+        if (Window::getKey(GLFW_KEY_EQUAL) == GLFW_PRESS)
+            FPS += (FPS < 5) ? 1 : ((FPS < 20) ? 5 : ((FPS < 60) ? 10 : 0));
+        if (Window::getKey(GLFW_KEY_MINUS) == GLFW_PRESS)
+            FPS -= (FPS > 20) ? 10 : ((FPS > 5) ? 5 : ((FPS > 1) ? 1 : 0));
+    }
+
+    game->update(dt);
 }
 
 void draw() {
-	glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
-	game->draw();
+        GL_CHECK(glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT));
+        game->draw();
 }
 
 int main(int argc, char** argv) {
-	GLFWmanager glfw(800, 600);
-	GLEWmanager glew;
-
-	if (DEBUG)
-		initDebug();
-	init();
-	CHECK_GL_ERR;
-	while (!glfwWindowShouldClose(Window::window)) {
-		update();
-		draw();
-		CHECK_GL_ERR;
-		glfwSwapBuffers(Window::window);
-		glfwPollEvents();
-	}
-
-	return 0;
-}
\ No newline at end of file
+    glfw = make_unique<GLFWmanager>();
+    glew = make_unique<GLEWmanager>();
+
+    glfw->init(800, 600);
+    glew->init();
```

Comment:
I would either go for the stack construction, (with init method) or the unique_ptr method, (with constructor) but not both. It's a little excessive that way imo